ament_cmake_core/cmake/index/ament_index_register_package.cmake

```diff
@@ -29,5 +29,5 @@ function(ament_index_register_package)
   endif()
 
   # register package name with the resource type 'packages'
-  ament_index_register_resource("packages" ${ARGN})
+  ament_index_register_resource("packages" ${ARG_PACKAGE_NAME})
```

Comment:
A few lines above it errors out if there is anything in `ARGN`. I don't see how this could ever pass something along. You're right that it won't work as is though. Should I just add the `PACKAGE_NAME` keyword argument or should I remove the check about `ARGN`?